camelot/view/mainwindow.py

```diff
@@ -34,30 +34,39 @@
 
 from camelot.view.controls.busy_widget import BusyWidget
 
-class MainWindow(QtWidgets.QMainWindow):
+class MainWindow(QtCore.QObject):
     """Main window of a Desktop Camelot application
     
     :param gui_context: an :class:`camelot.admin.action.application_action.ApplicationActionGuiContext`
         object
     :param parent: a :class:`QtWidgets.QWidget` object or :class:`None` 
+    :param window: a :class:`QtWidgets.QMainWindow` object or :class:`None`
     
+    If window is None, a new QMainWindow will be created.
     """
 
-    def __init__(self, gui_context, parent=None):
+    def __init__(self, gui_context, parent=None, window=None):
         from .workspace import DesktopWorkspace
         logger.debug('initializing main window')
-        QtWidgets.QMainWindow.__init__(self, parent)
+        QtCore.QObject.__init__(self)
+
+        if window is None:
+            self.window = QtWidgets.QMainWindow(parent)
+        else:
+            self.window = window
+            self.window.setParent(parent)
```

Comment:
is die 'reparenting' nodig, ik vrees dat we hiermee dependencies circles gaan maken tss python-c++

terwijl we die dependency er eigenlijk uit willen ...

Reply:
Ik heb deze call verwijderd.